src/niswitch/system_tests/test_system_niswitch.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/python
+
+import niswitch
+import pytest
+import warnings
+
+
+@pytest.fixture(scope='function')
+def session():
+    with niswitch.Session('', False, True, 'Simulate=1, DriverSetup=topology:2737/2-Wire 4x64 Matrix') as simulated_session:
+        yield simulated_session
+
+
+# Basic Use Case Tests
+def test_relayclose(session):
+    relay_name = 'kr0c0'
+    assert session.get_relay_position(relay_name) == niswitch.RelayPosition.OPEN
+    session.relay_control(relay_name, niswitch.RelayAction.CLOSE_RELAY)
+    assert session.get_relay_position(relay_name) == niswitch.RelayPosition.CLOSED
+    assert relay_name
+
+
+def test_channel_connection(session):
+    channel1 = 'c0'
+    channel2 = 'r0'
+    assert session.can_connect(channel1, channel2) == niswitch.PathCapability.PATH_AVAILABLE
+    session.connect(channel1, channel2)
+    assert session.can_connect(channel1, channel2) == niswitch.PathCapability.PATH_EXISTS
+    session.disconnect(channel1, channel2)
+    assert session.can_connect(channel1, channel2) == niswitch.PathCapability.PATH_AVAILABLE
+    session.connect(channel1, channel2)
+    assert session.can_connect(channel1, channel2) == niswitch.PathCapability.PATH_EXISTS
+    session.disconnect_all()
+    assert session.can_connect(channel1, channel2) == niswitch.PathCapability.PATH_AVAILABLE
+
+
+# Attribute Tests
+def test_viboolean_attribute(session):
```

Comment:
vi_boolean... this is consistent with what our code-generator produces.

This applies to other types.

Reply:
fixed